packages/core/src/lib/select/searchable-select.svelte

```diff
@@ -191,9 +245,16 @@ const handleKeydown = createHandleKey({
   },
   Escape: () => {
     if (menuState === CLOSED) {
-      value = '';
+      searchValue = '';
+      handleSelect(undefined);
+    } else {
+      // if the menu is open and we haven't selected a option (but we have pressed escape), we should
+      // select that to update the state accordingly
+      if (selectedSearchOption === undefined) {
+        handleSelect(undefined);
+      }
```

Comment:
This feels wrong; `Escape` should "abort" when the menu is open, it shouldn't change the value unless the popup is closed

https://www.w3.org/WAI/ARIA/apg/patterns/combobox/

> `Escape`: Dismisses the popup if it is visible. Optionally, if the popup is hidden before `Escape` is pressed, clears the combobox.